### PR TITLE
Delete avatar if a user is deleted

### PR DESCRIPTION
**Pull request by juliusknorr**
Make sure avatars are properly cleaned up if a user is deleted

### CODE REVIEW

**Review by rullzer — APPROVED** [review]

File: lib/private/Avatar/AvatarManager.php [hunk 1]
```diff
@@ -125,6 +126,20 @@ public function clearCachedAvatars() {
 		}
 	}
 
+	/**
+	 * @param string $userId
+	 * @throws NotPermittedException
```
Comment (kesselb) [inline lib/private/Avatar/AvatarManager.php]: What happens if that exception is thrown? Looks danger to me from the code to interrupt the delete process at this stage.

Comment (juliushaertl) [reply lib/private/Avatar/AvatarManager.php]: Ah yes, catch and error logging added for that. :+1:

File: lib/private/Avatar/AvatarManager.php [hunk 2]
```diff
@@ -125,6 +126,21 @@ public function clearCachedAvatars() {
 		}
 	}
 
+	/**
+	 * @param string $userId
+	 */
+	public function deleteUserAvatar(string $userId) {
```
Comment (nickvergessen) [inline lib/private/Avatar/AvatarManager.php]:
```suggestion
	public function deleteUserAvatar(string $userId): void {
```

File: lib/private/Avatar/AvatarManager.php [hunk 3]
```diff
@@ -125,6 +126,21 @@ public function clearCachedAvatars() {
 		}
 	}
 
+	/**
+	 * @param string $userId
```
Comment (nickvergessen) [inline lib/private/Avatar/AvatarManager.php]: remove the empty docs

**Comment by rullzer** [comment]
@juliushaertl mind to address the comments?

**Review by nickvergessen — APPROVED** [review]

File: lib/private/EventDispatcher/SymfonyAdapter.php [hunk 4]
```diff
@@ -63,10 +63,10 @@ public function dispatch($eventName, $event = null) {
 			$this->eventDispatcher->dispatch($eventName, $event);
 		} else {
 			// Legacy event
-			$this->logger->info(
-				'Deprecated event type for {name}: {class}',
-				[ 'name' => $eventName, 'class' => is_object($event) ? get_class($event) : 'null' ]
-			);
+//			$this->logger->info(
```
Comment (rullzer) [inline lib/private/EventDispatcher/SymfonyAdapter.php]: uhm?

Comment (nickvergessen) [reply lib/private/EventDispatcher/SymfonyAdapter.php]: Dumdidum, was not supposed to be commited

Comment (nickvergessen) [reply lib/private/EventDispatcher/SymfonyAdapter.php]: Fixed

**Review by blizzz — APPROVED** [review]

**Comment by blizzz** [comment]
Now you have failing tests though


```
...
/drone/src/lib/private/User/User.php:243 
...
```

**Comment by juliushaertl** [comment]
Let me restart, since those run fine locally.

**Comment by juliushaertl** [comment]
Let's see if that works.

**Review by georgehrke — APPROVED** [review]

**Pull request merged**